### DISCUSSION TITLE
Feature: View Insights for any GitHub topic

### DISCUSSION THREAD

**Discussion by Brian-Pob** [open-sauced/feedback#25] [General feedback or Bugs]
### Type of feature

🍕 Feature

### Current behavior

I'm new to OpenSauced Insights and I noticed that there was a very limited number of topics I could get insights from.
<img width="215" alt="CleanShot 2023-08-16 at 08 27 29" src="https://github.com/open-sauced/insights/assets/29676413/06d1d996-62d0-4f23-a607-5599f476b3fa">

Though I could get insights on more topics simply by manually changing the URL. This shows that the site can support more than the selected topics.

![CleanShot 2023-08-16 at 08 29 20](https://github.com/open-sauced/insights/assets/29676413/2c742c9e-6a0a-4b96-8615-3537611b720a)

### Suggested solution

Add the ability to manually search for topics to get insights. This can be done by having an input in the topics dropdown menu.

I would like to work on this issue if it gets accepted.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

**Comment by bdougie** [comment]
Open to UI suggestions on this. As you discovered we are limited to the finding other or new topics, the dropdown is the limiting factor. I don't know if there is an API endpoint for just topics, but that would be helpful.

We could potentially a list of all available topics from GitHub and search using some search tooling. 

## fuse.js for search
Another option is using existing setup, but allow users to add to the list. We can search list using [fuse.js](https://www.fusejs.io/). Fuse would work for a topics endpoint too.

**Comment by jpmcb** [comment]
Insight's uses the `v1/repos/search` API endpoint the `topic` query can be anything. For example, here's a curl query to get the top Flutter repos:

```bash
curl -X 'GET' \
  'https://api.opensauced.pizza/v1/repos/search?page=1&limit=10&range=30&topic=flutter' \
  -H 'accept: application/json'
```

So, yes, I believe it would only be a limitation on how the UI/UX for entering that `topic` API query would work

**Comment by bdougie** [comment]
My suggestion is instead of a search we open a command pallette (not designed) to allow the user to filter there instead. This will give us the ability to track to most request topics and only show the most popular, but show other topics in the command pallette.

Similar to github and other project command palettes. 

![Screen Shot 2023-08-22 at 4 24 55 PM](https://github.com/open-sauced/feedback/assets/5713670/77f7689e-4448-4b46-91f2-6d008e6e5bd1)
